.github/scripts/rust/setup.py

```diff
@@ -35,10 +28,22 @@ def generate_diffs():
 
 def find_local_crates():
     """
-    Returns all available crates in the workspace
+    Returns all available crates in the workspace.
+
+    If the crate is in a sub-crate of another crate, only the super-crate will be returned.
     :return: a list of crate paths
     """
-    return [path.relative_to(CWD).parent for path in CWD.rglob("Cargo.toml")]
+    all_crates = [path.relative_to(CWD).parent for path in CWD.rglob("Cargo.toml")]
+    checked_crates = []
+    for crate in all_crates:
+        seen = False
+        for other_crate in all_crates:
+            if str(crate).startswith(str(other_crate)) and crate != other_crate:
+                seen = True
+                break
+        if not seen:
+            checked_crates.append(crate)
+    return checked_crates
```

Comment:
This seems a little flakey, we're dealing with paths, so I'd generally prefer we use Pathlib and its associated functions, but also I'm not entirely sure as to why we're doing this in the first place.

It'd be good to have reasonings attached to the docstrings in places where you say things like
> If the crate is in a sub-crate of another crate, only the super-crate will be returned.

But also can we fix this by just not doing a recursive search or something?

Reply:
I simplified the logic to and elaborated on the docs: 64d5f16e18dd451f8fde0f7f07493deca5d2152c